verisignmdns/src/main/java/denominator/verisignmdns/VerisignMDNSResourceRecordSetApi.java

```diff
@@ -0,0 +1,156 @@
+        if (tempSet != null && tempSet.size() > 0) {
+            result = tempSet.first();
+        }
+        return result;
+    }
+
+    @Override
+    public void put(ResourceRecordSet<?> rrset) {
+
+        checkNotNull(rrset, "Resource Record was null");
+        checkNotNull(rrset.name(), "Resource Record Name was null");
+        checkNotNull(rrset.type(), "Resource Record Type was null");
+
+        // At this point disable delete
+        // as we might delete multiple records for name and type.
+        // Need to decide if that is correct
+        // ResourceRecordSet<?> rrsMatch = getByNameAndType(rrset.name(),
+        // rrset.type());
+        // if (rrsMatch != null) {
+        // deleteByNameAndType(rrset.name(), rrset.type());
+        // }
+        int ttlInt = 86000;
+        Integer ttlRRSet = rrset.ttl();
+        if (ttlRRSet != null) {
+            ttlInt = ttlRRSet.intValue();
+        }
+        String rData = getRDataStringFromRRSet(rrset);
+        api.createResourceRecord(domainName, rrset.type(), rrset.name(), "" + ttlInt, rData);
+
+    }
+
+    @Override
+    public void deleteByNameAndType(String name, String type) {
+        List<Record> recordList = api.getResourceRecordsListForTypeAndName(domainName, type, name);
+        if (recordList != null && !recordList.isEmpty()) {
+            // delete all records in recordList
+            for (Record record : recordList) {
+                api.deleteRecourceRecord(domainName, record.id);
+            }
+
+        } else {
+            throw new RuntimeException("deleteByNameAndType() failled to delete record for domain :" + domainName
+                    + " type :" + type + " No Record Found");
+        }
+    }
+
+    public static final class Factory implements denominator.ResourceRecordSetApi.Factory {
+        private Map<Zone, SortedSet<ResourceRecordSet<?>>> records;
+        private String domainName;
+        private VrsnMdns api;
+        private Provider<Credentials> credentialsProvider;
+
+        // unbound wildcards are not currently injectable in dagger
+        @SuppressWarnings({ "rawtypes", "unchecked" })
+        @Inject
+        Factory(Provider<Credentials> credentialsProvider, denominator.Provider provider, VrsnMdns api) {
+            this.records = Map.class.cast(records);
+            String url = provider.url();
+            this.api = api;
+            this.credentialsProvider = credentialsProvider;
+        }
+
+        @Override
+        public ResourceRecordSetApi create(String idOrName) {
+            Zone zone = Zone.create(idOrName);
+            return new VerisignMDNSResourceRecordSetApi(credentialsProvider, idOrName, api);
+        }
+    }
+
+    private String getRDataStringFromRRSet(ResourceRecordSet rRSet) {
+        StringBuilder sb = new StringBuilder();
+        if (rRSet != null && rRSet.records() != null) {
```

Comment:
null checks add cruft and confusion. make sure that they are in fact needed. for example, `rRSet.records()` will never be null. ps. looking at other code, you'll notice we don't mix case `rRSet`